src/SFA.DAS.Employer.PR.Web/Controllers/RegisteredProvidersController.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Employer.PR.Domain.Interfaces;
+using SFA.DAS.Employer.PR.Domain.Models;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using SFA.DAS.Employer.PR.Web.Services;
+
+namespace SFA.DAS.Employer.PR.Web.Controllers;
+
+[Authorize]
+[Route("registeredProviders", Name = RouteNames.GetRegisteredProviders)]
+public class RegisteredProvidersController(IOuterApiClient outerApiClient, ICacheStorageService cacheStorageService) : Controller
+{
+    [HttpGet]
+    public async Task<IActionResult> GetRegisteredProviders([FromQuery] string query, CancellationToken cancellationToken)
+    {
+        query = query.Trim();
+        if (query.Length < 3) return Ok(new List<RegisteredProvider>());
+
+        List<RegisteredProvider> providers = (await GetRegisteredProviders(cancellationToken))!;
+        var matchedProviders = providers
+            .Where(provider => provider.Name.Contains(query, StringComparison.OrdinalIgnoreCase)
```

Comment:
Where is the order by being set?

Reply:
I never set the order on this, but can do it alphabetically when added to the cache.

Reply:
will do in follow up

Reply:
yes that would help as we have to list it alphabetically right?